dlib/dnn/loss.h

```diff
@@ -1240,7 +1240,7 @@ namespace dlib
 
                 hit_truth_table.assign(hit_truth_table.size(), false);
                 final_dets.clear();
-
+                truth_idxs.clear();
```

Comment:
Just move `truth_idxs` into the loop body so the `clear()` is unneeded. I'm not sure what I was thinking.  The formatting of the `truth_idxs` declaration is even weird :|

Reply:
Moved. (Didn't change formatting.)